lib/oauth2-client.js

```diff
@@ -143,34 +126,43 @@ async function _requestAccessToken({url, body, httpsAgent}) {
   return false;
 }
 
-async function getAccessTokenRetryAndExitOnFailure({
-  client_id, client_secret, token_endpoint, grant_type, scope, httpsAgent
+export async function getAuthzClients({clients}) {
+  const promises = [];
+  for(const client in clients) {
+    const clientInfo = clients[client];
+    promises.push(_createAccessToken({...clientInfo}));
```

Comment:
Any reason not to simplify to this?

```suggestion
    promises.push(_createAccessToken(clientInfo));
```

Reply:
Updated.